crates/data-transformer/src/reverse_transformer/mod.rs

```diff
@@ -0,0 +1,69 @@
+mod transform;
+mod types;
+
+use crate::reverse_transformer::transform::{ReverseTransformer, TransformationError};
+use crate::shared::extraction::extract_function_from_selector;
+use starknet::core::types::contract::{AbiEntry, AbiFunction};
+use starknet_types_core::felt::Felt;
+
+#[derive(Debug, thiserror::Error)]
+pub enum ReverseTransformError {
+    #[error(r#"Function with selector "{0}" not found in ABI of the contract"#)]
+    FunctionNotFound(Felt),
+    #[error(transparent)]
+    TransformationError(#[from] TransformationError),
+}
+
+/// Transforms a calldata into a Cairo-like string representation of the arguments
+pub fn reverse_transform_input(
```

Comment:
I assume this is just needed for backtrace? Since sncast will only need to transform function outputs.